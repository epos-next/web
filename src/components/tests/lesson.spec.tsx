import Lesson from "@components/lesson";
import UiHelper from "@helpers/ui-helper";
import { render } from "@testing-library/react";
import React from "react";

beforeEach(() => {
    jest.spyOn(UiHelper, "getLessonColor").mockReturnValue({
        colorAccent: "#123123",
        color: "#456456"
    })
})

afterEach(() => {
    jest.clearAllMocks()
})

it("should match snapshot", () => {
    const { asFragment } = render(<Lesson subject="Физика" />)
    expect(asFragment()).toMatchSnapshot()
});

it("should use subject prop", () => {
    const { getByTestId, rerender } = render(<Lesson subject="Английский Язык" />)
    expect(getByTestId("lesson-name")).toHaveTextContent("Английский Язык")

    rerender(<Lesson subject="Геометрия" />)
    expect(getByTestId("lesson-name")).toHaveTextContent("Геометрия")

    rerender(<Lesson subject="Какая-то фигня" />)
    expect(getByTestId("lesson-name")).toHaveTextContent("Какая-то фигня")
});

it("should use subtitle prop", () => {
    const { getByTestId, rerender } = render(<Lesson subject="ИЗО" subtitle="123" />)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("123")

    rerender(<Lesson subject="ИЗО" subtitle="hello my brooo" />)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("hello my brooo")

    rerender(<Lesson subject="ИЗО" subtitle="4 параграф упр 23" />)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("4 параграф упр 23")
});

it("should use className prop", () => {
    const { getByTestId, rerender } = render(<Lesson subject="ИЗО" className="123" />)
    expect(getByTestId("lesson-container")).toHaveClass("123")

    rerender(<Lesson subject="ИЗО" className="hello my brooo" />)
    expect(getByTestId("lesson-container")).toHaveClass("hello my brooo")

    rerender(<Lesson subject="ИЗО" className="4 параграф упр 23" />)
    expect(getByTestId("lesson-container")).toHaveClass("4 параграф упр 23")
});

it("should use correct letter avatar", () => {
    const { getByTestId, rerender } = render(<Lesson subject="Английский Язык" />)
    expect(getByTestId("avatar")).toHaveTextContent("А")
    expect(UiHelper.getLessonColor).toHaveBeenCalledWith("Английский Язык")

    rerender(<Lesson subject="Геометрия" />)
    expect(getByTestId("avatar")).toHaveTextContent("Г")
    expect(UiHelper.getLessonColor).toHaveBeenCalledWith("Геометрия")

    rerender(<Lesson subject="Какая-то фигня" />)
    expect(getByTestId("avatar")).toHaveTextContent("К")
    expect(UiHelper.getLessonColor).toHaveBeenCalledWith("Какая-то фигня")
});
