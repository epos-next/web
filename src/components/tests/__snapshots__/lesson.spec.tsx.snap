// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  .c3 {
  font-size: 17px;
  line-height: 20px;
  color: var(--primary);
}

.c4 {
  font-size: 15px;
  line-height: 18px;
  color: var(--secondary);
}

.c1 {
  width: 36px;
  height: 36px;
  border-radius: 50%;
  background-color: #456456;
  color: #123123;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-weight: 500;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 15px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    class="c0 lesson"
    data-testid="lesson-container"
  >
    <div
      class="c1"
      color="#456456"
      data-testid="avatar"
    >
      Ф
    </div>
    <div
      class="c2"
    >
      <span
        class="c3"
        data-testid="lesson-name"
      >
        Физика
      </span>
      <span
        class="c4"
        data-testid="lesson-subtitle"
      />
    </div>
  </div>
</DocumentFragment>
`;
